components/txn_types/src/lock.rs

```diff
@@ -347,6 +347,14 @@ impl Lock {
             return Ok(());
         }
 
+        // When ts is `TimeStamp::max()` and `bypass_locks` is not empty, the lock
+        // read for the first time has been resolved (or whose min_commit_ts has been
+        // advanced). And because other new locks should not block such reading requests,
+        // we can ignore all seen locks in this case.
+        if ts == TimeStamp::max() && !bypass_locks.is_empty() {
```

Comment:
So  this doesn't take effect when the first lock it meets is an async commit lock whose  min_commit_ts can't be pushed?

Reply:
🤔 Looks strange. As what you said, this PR should not have much improvement, since I didn't modify TiDB to put resolved async commit locks in it.

Reply:
What's the conclusion?

Reply:
Use https://github.com/pingcap/tidb/pull/24084 to implement it instead.

With that solution, it is possible (but rare) that the retries keep encountering different locks. If we want to solve this problem, I think a explicit protocol is preferred instead of reusing `bypass_locks`.